daemon/list.go

```diff
@@ -292,6 +292,10 @@ func (daemon *Daemon) transformContainer(container *Container, ctx *listContext)
 		Names:   ctx.names[container.ID],
 		ImageID: container.ImageID,
 	}
+	if newC.Names == nil {
+		// Dead containers will often have no name, so make sure the response isn't  null
```

Comment:
Some nitpicking: `s/  / /` in case you amend this anyway.  Also dots at the end of comments are nice.